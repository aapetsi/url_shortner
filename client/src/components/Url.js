import React from 'react'

const Url = (props) => {
  const { id, originalUrl, shortUrl } = props
  console.log(props)
  return (
    <div>
      <p>https://pbid.io/{shortUrl}</p>
    </div>
  )
}

export default Url
