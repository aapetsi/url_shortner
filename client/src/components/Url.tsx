import React, {FunctionComponent} from 'react'
import Axios from 'axios'

export type UrlProps =  {
  key: string,
  shortUrl: string,
  id: string,
  originalUrl: string,
  shortUrlHash: string
}

const Url: FunctionComponent<UrlProps> = ({id, originalUrl, shortUrl, shortUrlHash, key}) => {

  const handleOpenLink = () => {
    window.open(originalUrl, '_blank')
  }

  return (
    <div data-testid="url">
      <p
        className='link'
        onClick={handleOpenLink}
      >
        {shortUrl}
      </p>
    </div>
  )
}

export default Url
