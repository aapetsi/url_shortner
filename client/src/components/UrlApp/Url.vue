<template>
  <tr>
    <td class="link" data-testid="originalUrl">{{ originalUrl }}</td>
    <td class="link" data-testid="shortUrl">{{ shortUrl }}
      <button class="copy-clipboard" @click="handleCopyClipboard">Copy</button>
    </td>
    <td>
      <button class="delete-one" @click="handleDeleteUrl">Delete</button>
    </td>
  </tr>
</template>

<script lang="ts">
import Vue from 'vue'
import { mapActions } from 'vuex'

export default Vue.extend({
  props: {
    shortUrl: {
      type: String,
      required: true
    },
    id: {
      type: String,
      required: true
    },
    originalUrl: {
      type: String,
      required: true
    },
    shortUrlHash: {
      type: String,
      required: true
    }
  },
  name: 'Url' as string,
  data: () => ({}),
  methods: {
    ...mapActions({
      handleDelete: 'urls/deleteUrl'
    }),
    handleDeleteUrl() : void {
      this.handleDelete(this.id)
    },

    handleCopyClipboard() : void {
      navigator.clipboard.writeText(this.shortUrl)
      alert('copied to clipboard')
    }
  }
})
</script>

<style scoped>
  .copy-clipboard {
    display: inline-block;
    background: green;
    color: white;
    height: 32px;
    margin-left: 10px;
    width: 60px;
    font-weight: bold;
    text-transform: uppercase;
    border-radius: 5px;
    border: 0;
    box-shadow: none;
    cursor: pointer;
  }

  .delete-one {
    height: 32px;
    background: #dc3545;
    color: white;
    font-weight: bold;
    text-transform: uppercase;
    border-radius: 5px;
    border: 0;
    box-shadow: none;
    text-align: center;
    cursor: pointer;
  }

  button:hover {
    background: #556476;
  }

  .link {
    color: blue;
    cursor: pointer;
    text-decoration: underline;
  }

  @media only screen and (max-width: 540px) {
    .copy-clipboard {
      display: none;
    }

    .delete-one {
      display: none;
    }
  }
</style>