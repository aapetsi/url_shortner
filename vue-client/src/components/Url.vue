<template>
  <p>Url Component</p>
</template>

<script>
export default {
  name: 'Url',
  data: function() {
    return {

    }
  }
}
</script>

<style scoped>

</style>