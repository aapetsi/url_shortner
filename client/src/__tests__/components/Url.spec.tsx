import React from 'react'
import Url, {UrlProps} from '../../components/Url'
import {render, fireEvent, waitForElement} from '@testing-library/react'

describe('<Url />', () => {
  let defaultProps : {
    key: string,
    shortUrl: string,
    id: string,
    originalUrl: string,
    shortUrlHash: string
  }
  
  test('should match snapshot', async () => {
    const {findByTestId} = render(<Url {...defaultProps}/>)
    const url = await findByTestId('url')
    expect(url).toMatchSnapshot()
  })
})