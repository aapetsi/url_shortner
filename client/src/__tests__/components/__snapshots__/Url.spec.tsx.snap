// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Url /> should match snapshot 1`] = `
<div
  data-testid="url"
>
  <p
    class="link"
  />
</div>
`;
